src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -1620,7 +1620,12 @@ impl<T: AstInfo> AstDisplay for CreateSecretStatement<T> {
         }
         f.write_node(&self.name);
         f.write_str(" AS ");
-        f.write_node(&self.value);
+
+        if f.simple() {
```

Comment:
I don't see where this is triggering. The new code in statement_logging.rs explicitly calls `to_ast_string_redacted()` for SECRETs, so the values are redacted there.

Reply:
Yeah this would also trigger for cases where we use `.to_ast_string()`, which I thought would also help de-risk tracing. I incorrectly assumed we didn't guarantee round-tripping for `to_ast_string()` and thought we only did for `to_ast_string_stable()`, but some tests failed that proved otherwise!

I just changed this to only redact when formatting as redacted which is good enough for this PR. And I figured we can re-visit the guarantees of `to_ast_string()` without time pressure